src/libraries/Common/src/Interop/OSX/Interop.libSystem.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class libSystem
+    {
+        [LibraryImport(Interop.Libraries.libSystem)]
+        public static unsafe partial int mach_timebase_info(mach_timebase_info_data_t* info);
+        public struct mach_timebase_info_data_t
```

Comment:
@jkoritzinsky @AaronRobinsonMSFT In terms of marshaller best practices, do we need to explicitly specify sequential layout for such structs?

```suggestion
        [StructLayout(LayoutKind.Sequential)]
        public struct mach_timebase_info_data_t
```

Reply:
Technically, no. Value types in .NET default to sequential layout. However, and this is annoying, there are some Roslyn warnings that are suppressed if one does explicitly mark the type with sequential layout. The reasoning here is historical, but the gist is if Roslyn complains about unreferenced fields, which can happen for types used in interop, then placing `StructLayout(LayoutKind.Sequential)` on the type will automatically suppress the warning.

The interop team's general guidance here has been to accept the defaults except where there is annoying friction with C# or where the tooling requires explicit details. This falls into the C# friction bucket, but only if a warning is emitted.

Reply:
@AaronRobinsonMSFT thank you for a very detailed answer!

Reply:
I can't see any related warnings in the compilation (and also we seem to have warnings as errors enabled in this part?).

Does this mean this attribute is unnecessary? I am totally okay with adding that if required. Though, yeah, we all know that sequential is the default struct layout 😅

Reply:
> Does this mean this attribute is unnecessary?

Yes.